test/ng/compileSpec.js

```diff
@@ -12111,6 +12141,626 @@ describe('$compile', function() {
+      }));
+
+      it('should accept trusted values for non-whitelisted values', inject(function($rootScope, $compile, $sce) {
+        $rootScope.testUrl = $sce.trustAsUrl('javascript:foo()'); // `javascript` is not whitelisted
+        element = $compile('<a ng-prop-href="testUrl"></a>')($rootScope);
+        $rootScope.$digest();
+        expect(element.prop('href')).toEqual('javascript:foo()');
+
+        element = $compile('<a ng-prop-href="testUrl"></a>')($rootScope);
+        $rootScope.$digest();
+        expect(element.prop('href')).toEqual('javascript:foo()');
+      }));
+
+      it('should sanitize non-whitelisted values', inject(function($rootScope, $compile) {
+        $rootScope.testUrl = 'javascript:foo()'; // `javascript` is not whitelisted
+        element = $compile('<a ng-prop-href="testUrl"></a>')($rootScope);
+        $rootScope.$digest();
+        expect(element.prop('href')).toEqual('unsafe:javascript:foo()');
+
+        element = $compile('<a ng-prop-href="testUrl"></a>')($rootScope);
+        $rootScope.$digest();
+        expect(element.prop('href')).toEqual('unsafe:javascript:foo()');
+      }));
+
+      it('should not sanitize href on elements other than anchor', inject(function($compile, $rootScope) {
+        element = $compile('<div ng-prop-href="testUrl"></div>')($rootScope);
+        $rootScope.testUrl = 'javascript:doEvilStuff()';
+        $rootScope.$apply();
+
+        expect(element.prop('href')).toBe('javascript:doEvilStuff()');
+      }));
+
+      it('should not sanitize properties other then those configured', inject(function($compile, $rootScope) {
+        element = $compile('<a ng-prop-title="testUrl"></a>')($rootScope);
+        $rootScope.testUrl = 'javascript:doEvilStuff()';
+        $rootScope.$apply();
+
+        expect(element.prop('title')).toBe('javascript:doEvilStuff()');
+      }));
+
+      it('should use $$sanitizeUri', function() {
+        var $$sanitizeUri = jasmine.createSpy('$$sanitizeUri').and.returnValue('someSanitizedUrl');
+        module(function($provide) {
+          $provide.value('$$sanitizeUri', $$sanitizeUri);
+        });
+        inject(function($compile, $rootScope) {
+          element = $compile('<a ng-prop-href="testUrl"></a>')($rootScope);
+          $rootScope.testUrl = 'someUrl';
+          $rootScope.$apply();
+          expect(element.prop('href')).toMatch(/^http:\/\/.*\/someSanitizedUrl$/);
+          expect($$sanitizeUri).toHaveBeenCalledWith($rootScope.testUrl, false);
+
+          $$sanitizeUri.calls.reset();
+
+          element = $compile('<a ng-prop-href="testUrl"></a>')($rootScope);
+          $rootScope.$apply();
+          expect(element.prop('href')).toMatch(/^http:\/\/.*\/someSanitizedUrl$/);
+          expect($$sanitizeUri).toHaveBeenCalledWith($rootScope.testUrl, false);
+        });
+      });
+
+      it('should not have endless digests when given arrays in concatenable context', inject(function($compile, $rootScope) {
```

Comment:
Didn't we agree that ng-prop values were not concatenable?

Reply:
I thought this one might be worth it still, although could probably be renamed. I think I mainly wanted to test that using an object literal doesn't cause an `infdig`.